crates/net/discv4/src/config.rs

```diff
@@ -247,6 +250,12 @@ impl Discv4ConfigBuilder {
         self
     }
 
+    /// Sets the discovery protocol to be disabled if disable_discovery is true.
+    pub fn set_discovery(&mut self, disable_discovery: bool) -> &mut Self {
+        self.config.disable_discovery = disable_discovery;
+        self
+    }
+
```

Comment:
I feel like this is the wrong place here, and should be moved to networkconfig.

Because you don't want to configure a Config and then disable it.